wrappers/adaptive/popcorn.HTMLAdaptiveMediaElement.js

```diff
@@ -151,19 +157,35 @@
                     hls.loadSource(source);
                     hls.attachMedia(media);
                   } else if (media.canPlayType('application/vnd.apple.mpegurl')) {
-                    var sources = media.getElementsByTagName('source');
-                    if(source && source !== sources[0].src) {
-                      sources[0].src = source;
-                      media.load();
+                    // IE and Edge do not understand source setting here for MSE BLOB
+                    if (isMicrosoftBrowser()) {
+                      if(source && source !== media.getAttribute('src')) {
+                        media.setAttribute('src', source);
+                        media.load();
+                      }
+                    } else {
+                      var sources = media.getElementsByTagName('source');
+                      if(source && source !== sources[0].src) {
+                        sources[0].src = source;
+                        media.load();
+                      }
                     }
                   }
                 });
                 break;
               default:
-                var sources = media.getElementsByTagName('source');
-                if(source && source !== sources[0].src) {
-                  sources[0].src = source;
-                  media.load();
+                // IE and Edge do not understand source setting here for MSE BLOB
+                if (isMicrosoftBrowser()) {
```

Comment:
I see identical code. May be you'll create a new function?

Reply:
yeah let's do that.